### ISSUE TITLE
Incorrect layer name from path

### ISSUE THREAD

**Issue by maxdreyer** [rachtibat/zennit-crp#8]
Hi all,

in **_maximization.py_** in line 110 of method **collect_results**, the layer name is extracted from the path as

```
for path in path_list:
    filename = path.split("/")[-1]
    l_name = filename.split("_")[0]
```

Here, the ending of **filename** consists of a layer's name and some numbers (which are separated by " _ ").
This, however, assumes that the layer's name does not consist of " _ ", which can happen. In this case, the analysis does not work.

**Comment by rachtibat** [comment]
Hi Max,

Thanks for your numerous issues. I'll be taking a closer look at them in the near future.
Regarding this issue: You're right, this could be a problem.

Greetings

**Comment by rachtibat** [comment]
I think, we can solve this by using regular expression. The latter part is always in the form [0-9]+_[0-9]+. This could be exploited.

**Comment by rachtibat** [comment]
fixed with d5337861a4bdd3ed18dd764ac5362c54a3a87b2f
